File: petties_mobile/lib/data/models/user_response.dart
import 'base_model.dart';

/// User response model from API
class UserResponse extends BaseModel {
  final String userId;
  final String username;
  final String email;
  final String? fullName;
  final String? phone;
  final String? avatar;
  final String role;
  final String? workingClinicId;
  final String? workingClinicName;
  final String createdAt;
  final String updatedAt;

  UserResponse({
    required this.userId,
    required this.username,
    required this.email,
    this.fullName,
    this.phone,
    this.avatar,
    required this.role,
    this.workingClinicId,
    this.workingClinicName,
    required this.createdAt,
    required this.updatedAt,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      userId: json['userId'] ?? '',
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      fullName: json['fullName'],
      phone: json['phone'],
      avatar: json['avatar'],
      role: json['role'] ?? '',
      workingClinicId: json['workingClinicId'],
      workingClinicName: json['workingClinicName'],
      createdAt: json['createdAt'] ?? '',
      updatedAt: json['updatedAt'] ?? '',
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'username': username,
      'email': email,
      'fullName': fullName,
      'phone': phone,
      'avatar': avatar,
      'role': role,
      'workingClinicId': workingClinicId,
      'workingClinicName': workingClinicName,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }
}

File: backend-spring/petties/src/main/java/com/petties/petties/dto/auth/UserResponse.java
package com.petties.petties.dto.auth;

import com.petties.petties.model.enums.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {

    private UUID userId;
    private String username;
    private String email;
    private String fullName;
    private String phone;
    private String avatar;
    private Role role;
    private UUID workingClinicId; // For CLINIC_MANAGER and VET
    private String workingClinicName; // Clinic name for display

    // VET-specific fields
    private String specialty; // VET_GENERAL, VET_SURGERY, VET_DENTAL, VET_DERMATOLOGY, GROOMER
    private java.math.BigDecimal ratingAvg; // Rating trung bình (1.0 - 5.0)
    private Integer ratingCount; // Số lượt đánh giá

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
